MotorControl/MotorControl/iOS/ResultObjects/MCTTappingResultObject.swift

```diff
@@ -191,3 +278,109 @@ public struct MCTTappingSample : SampleRecord, Codable {
+        case .startDate, .endDate:
+            return .init(propertyType: .format(.dateTime))
+        case .viewSize:
+            return .init(propertyType: .primitive(.string), propertyDescription:
+                            "The size of the bounds of the step view containing the tap targets.")
+        case .buttonRectLeft, .buttonRectRight:
+            return .init(propertyType: .primitive(.string), propertyDescription:
+                            "The frame of the buttons relative to the step view bounds.")
+        case .tapCount:
+            return .init(propertyType: .primitive(.integer), propertyDescription:
+                            "The tap count of hits that were within the buttons.")
+        case .samples:
+            return .init(propertyType: .referenceArray(MCTTappingSample.documentableType()), propertyDescription:
+                            "An array of collected tapping samples.")
+            
+        }
+    }
+    
+    public static func examples() -> [MCTTappingResultObject] {
+        var result = MCTTappingResultObject(identifier: "tapping")
+        result.buttonRect1 = .init(x: 60, y: 500, width: 40, height: 40)
+        result.buttonRect2 = .init(x: 160, y: 500, width: 40, height: 40)
+        result.stepViewSize = .init(width: 320, height: 640)
+        result.samples = MCTTappingSample.examples()
+        result.tapCount = 1
+        return [result]
+    }
+}
+
+extension MCTTappingResultObject : DocumentableRootObject {
+    public var jsonSchema: URL {
+        .init(string: "TappingResultObject.json", relativeTo: kSageJsonSchemaBaseURL)!
+    }
+    
+    public var documentDescription: String? {
+        "The `TappingResultObject` records the results of a tapping interval test."
+    }
+}
+
+extension MCTTappingSample : DocumentableStruct {
+    public static func codingKeys() -> [CodingKey] {
+        CodingKeys.allCases
+    }
+    
+    public static func isRequired(_ codingKey: CodingKey) -> Bool {
+        guard let key = codingKey as? CodingKeys else { return false }
+        return key != .timestamp
+    }
+    
+    public static func documentProperty(for codingKey: CodingKey) throws -> DocumentProperty {
+        guard let key = codingKey as? CodingKeys else {
+            throw DocumentableError.invalidCodingKey(codingKey, "\(codingKey) is not recognized for this class")
+        }
+        switch key {
+        case .stepPath:
+            return .init(propertyType: .primitive(.string), propertyDescription:
+                            "The current step path.")
+        case .timestamp:
+            return .init(propertyType: .primitive(.number), propertyDescription:
+                            "A relative timestamp indicating the time of the tap event where `.zero` is the start of the recording.")
+        case .uptime:
+            return .init(propertyType: .primitive(.number), propertyDescription:
+                            "Clock time for the sample.")
```

Comment:
Should this say something more descriptive, like 'system clock time'? I tend to think of "clock time" meaning like a wall clock or a clock radio....